src/components/Select/Select.tsx

```diff
@@ -45,45 +58,105 @@ const Select = React.forwardRef<HTMLInputElement, Props & InputProps>(
       selectedOption = emptyValue,
       multi = false,
       options,
+      isAsync = false,
+      asyncSearch = () => {},
       status = 'normal',
+      minCharactersToSearch = 0,
+      highlightSearch = false,
+      isSearchable = true,
       ...restInputProps
     },
     ref
   ) => {
     const theme = useTheme();
     const [open, setOpen] = React.useState(false);
+    const [isLoading, setIsLoading] = React.useState(false);
     const [inputValue, setInputValue] = React.useState(defaultValue || selectedOption);
     const [searchValue, setSearchValue] = React.useState('');
 
     useEffect(() => {
       setInputValue(defaultValue || selectedOption);
     }, [defaultValue, selectedOption]);
 
+    useEffect(() => {
+      if (isAsync) {
+        setIsLoading(false);
+      }
+    }, [isAsync, options]);
+
     const handleOptionClick = (option: SelectOption) => {
       setInputValue(option);
       setOpen(false);
-      setSearchValue('');
+      if (isSearchable) {
+        setSearchValue('');
+      }
       handleSelectedOption(option);
     };
 
+    const handleOnKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+      // if backspace
+      if (e.keyCode === 8) {
```

Comment:
This is optional. Comments will be stale as soon as you write them.  Therefore I think it's better to have variables instead of comments for these cases. 

e.x.
```
const isBackspaceKey =  e.keyCode === 8 || e.which === 32;

if(isBackspaceKey) {...}
```

Reply:
This wasn't in the scope of this PR but yeah I totally agree with you. Changing it too